databricks_cli/pipelines/cli.py

```diff
@@ -59,8 +69,20 @@ def deploy_cli(api_client, spec_arg, spec):
         raise RuntimeError('The spec should be provided either by an option or argument')
     src = spec_arg if bool(spec_arg) else spec
     spec_obj = _read_spec(src)
+    if 'id' not in spec_obj:
+        pipeline_id = str(uuid.uuid4())
+        click.echo("Updating spec with id: {}".format(pipeline_id))
+        spec_obj['id'] = pipeline_id
+        _write_spec(src, spec_obj)
+    elif not _validate_pipeline_id(spec_obj['id']):
+        error_and_quit("Pipeline id has invalid characters")
     PipelinesApi(api_client).deploy(spec_obj)
 
+    pipeline_id = spec_obj['id']
+    base_url = "{0.scheme}://{0.netloc}/".format(urlparse(api_client.url))
+    pipeline_url = urljoin(base_url, "#joblist/pipelines/{}".format(pipeline_id))
```

Comment:
File a task on our side to return this from deployment (and then also update this to use that).

Reply:
Does this work for MT shards?

Reply:
Yes `api_client.url` cleans up the `?o=123` part and just returns the host url